sycl/plugins/unified_runtime/ur/adapters/hip/usm.cpp

```diff
@@ -106,15 +107,70 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMFree(ur_context_handle_t hContext,
       Result = UR_CHECK_ERROR(hipFree(pMem));
     }
     if (Type == hipMemoryTypeHost) {
-      Result = UR_CHECK_ERROR(hipFreeHost(pMem));
+      Result = UR_CHECK_ERROR(hipHostFree(pMem));
     }
   } catch (ur_result_t Error) {
-    return Error;
+    Result = Error;
   }
-  hContext->removeUSMMapping(pMem);
   return Result;
```

Comment:
I find early exits cleaner. Also note, that `UR_CHECK_ERROR` effectively terminates, so `Result` will only be altered in the `catch` block.

Reply:
Just to check if you are suggesting to avoid using `UR_CHECK_ERROR` or something else?

Reply:
Thanks for clarifying, that PR looks better and have gave it an approval.